packages/fx-core/src/core/index.ts

```diff
@@ -731,7 +734,9 @@ export class FxCore implements Core {
       );
       await fs.writeFile(
         path.join(inputs.projectPath!, `.gitignore`),
-        `node_modules\n/.${ConfigFolderName}/*.env\n/.${ConfigFolderName}/*.userdata\n.DS_Store\n${ArchiveFolderName}\n${ArchiveLogFileName}`
+        isMultiEnvEnabled()
```

Comment:
Is something like this easier to read and maintain?

```typescript
const ignoreFilePatterns = [
  "node_modules",
  `.${ConfigFolderName}/${InputConfigsFolderName}/${localSettingsFileName}`,
  `.${ConfigFolderName}/${PublishProfilesFolderName}/*.userdata`,
  ".DS_Store",
  ...
];
const content = ignoreFilePatterns.join("\n");
```

Reply:
Looks better. Updated~